common/src/types/token.rs

```diff
@@ -701,4 +704,15 @@ impl Token {
                 )
             })
     }
+
+    /// Gets the amount of the token as an f64, accounting for the associated
+    /// number of decimals.
+    ///
+    /// Note that due to conversion to f64, the result may lose precision.
+    pub fn get_amount_f64(&self, amount: u128) -> Result<f64, String> {
+        let decimals = self.get_decimals().ok_or(DECIMALS_NOT_FOUND_ERR.to_string())?;
```

Comment:
I'd suggest we just `unwrap_or_default` here, so that we get zero decimals. That way this method can be infallible